lib/BigML.js

```diff
@@ -160,24 +169,45 @@ BigML.prototype.request = function (options, retry, cb) {
       }
     }
     if (body) {
-      var json;
+      var json, safeId;
       try {
         if (process.env.BIGML_DEBUG || false) {
-          console.log(body);
+          logger.info(body);
         }
         json = JSON.parse(body);
         if (typeof self.storage !== 'undefined' && self.storage !== null &&
             reqOptions.method !== "DELETE"  && reqOptions.store &&
             typeof reqOptions.resource !== 'undefined') {
-          try {
-            if (!fs.existsSync(self.storage)) {
-              fs.mkdirSync(self.storage);
-            }
-            filename = path.join(self.storage,
-                                 options.resource.replace("/", "_"));
-            fileLength = fs.writeFileSync(filename, body);
-          } catch (ferr) {logger.error('Storage error: ' + ferr +
-                          "\nfilename: " + filename);}
+          safeId = options.resource.replace('/', '_');
+          // local storage in file system expects the path to a directory
+          if (typeof self.storage === 'string') {
+            try {
+              if (!fs.existsSync(self.storage)) {
+                fs.mkdirSync(self.storage);
+              }
+              filename = path.join(self.storage, safeId);
+              fileLength = fs.writeFileSync(filename, body);
+            } catch (ferr) {logger.error('Storage error: ' + ferr +
+                            "\nfilename: " + filename);}
+          } else if (typeof self.storage === 'object' &&
+                     typeof self.storage.set === 'function') {
+            // cache-like type of storage needs a set function to be provided
+            // expecting the following arguments:
+            // self.storage.set(key, val, cb). The ttl value should be
+            // defined in the cache manager constructor
+            self.storage.set(safeId, body,
+                             function(error) {
+              if (error) {logger.error('Problem caching ' + options.resource +
+                                       ': ' + error);}
+              if (process.env.BIGML_DEBUG || false) {
+                self.storage.get(safeId, function (error, data) {
+                  logger.info("Stored in cache:"+ JSON.parse(data).resource)});}});
+          } else {
+            logger.error("Storage can the path to a file system directory" +
+                         " used as storage or to a cache-manager object" +
+                         " that provides set, get and del methods. " +
+                         "None of these options has been found.");
```

Comment:
nit.  I would say: "Invalid storage.  Storage can be .... del methods." (and omit the last sentence)